addons/im_livechat/static/src/embed/cors/livechat_service_patch.js

```diff
@@ -0,0 +1,11 @@
+/* @odoo-module */
+
+import { patch } from "@web/core/utils/patch";
+import { LivechatService } from "../common/livechat_service";
+
+patch(LivechatService.prototype, {
+    leaveSession() {
+        this.busService.deleteChannel(`mail.guest_${this.guestToken}`);
```

Comment:
I'm still wondering whether we should delete the channel on leave, or if we should delete it when starting a new session.
Also wondering if we could somehow re-use the same guest if starting a new livechat and the page is still open?
Giving existing guest token to init if possible?

Reply:
I think you are right: for non CORS live chats, the guest is kept via the cookie mechanism. I don't see why it should behave differently in the CORS scenario. I adapted the PR to handle this case.